openedx/openedx-platform
# [ENT-18] Data sharing consent frontend

This pull request adds some frontend features to the data sharing consent feature. It provides a step in the pipeline where users who have not consented to data sharing can either provide consent or decline to do so. Declining to do so will result in the user being logged out.

**JIRA tickets**: Implements ENT-18

**Discussions**: Discussed in ENT-24

**Dependencies**: Depends on #13687 (commits duplicated in this branch)

**Sandbox URL**:
- **LMS:** http://pr13737.sandbox.opencraft.hosting
- **Admin:** http://pr13737.sandbox.opencraft.hosting/admin/third_party_auth (use staff account)

**Merge deadline**: EOD 20 October 2016

**Testing instructions**:
1. In the sandbox admin, verify that the TestShib SAML IdP is set to neither require nor request data sharing consent.
2. Register a new user, using the TestShib SAML IdP. Verify that the process is identical to the one that exists now.
3. Disconnect that user from the IdP and sign out.
4. Modify the TestShib SAML IdP setting to request data sharing consent.
5. Register a new user, using the TestShib SAML IdP, noting that a checkbox for consent is present, checking it.
6. In the admin panel, note that DataSharingConsentAudit object for this user notes that data sharing is permitted.
7. Disconnect this user from the IdP and sign out.
8. Register a new user, using the TestShib SAML IdP, noting that a checkbox for consent is present, but leaving it blank.
9. Sign out of this user, but do not disconnect it from the IdP.
10. In the admin panel, note that the DataSharingConsentAudit for this user notes that data sharing is not permitted.
11. In the admin panel, set the TestShib SAML IdP to require data sharing consent.
12. Attempt to sign in, using the TestShib SAML user account you registered previously. Note that it redirects you to a page stating that additional permissions are required. Do not grant permission, but click Submit, noting that you are logged out.
13. Again attempt to sign in, using the TestShib SAML user account you registered previously. Again note that you are redirected to a page stating that additional permissions are required. Check the box granting permissions and click Submit, noting that you are logged in successfully.
14. Disconnect the user from the IdP and sign out.
15. Attempt to register a new user using the TestShib SAML IdP. Note that the checkbox for data sharing consent is present; note that if you attempt to register a user without checking it, you encounter an error message.
16. Note that once registered, the OpenEdX platform behaves normally.

**Author notes and concerns**:
1. UI requires some additional scheming/styling; feedback on this process is welcome.

**Reviewers**
- [ ] @edx/devops 

**Settings**

``` yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
  ENABLE_THIRD_PARTY_AUTH: true
```
